checkstyle.xml

```diff
@@ -0,0 +1,352 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+  "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<!--
+    Checkstyle configuration that checks the Google coding conventions from Google Java Style
+    that can be found at https://google.github.io/styleguide/javaguide.html
+
+    Checkstyle is very configurable. Be sure to read the documentation at
+    http://checkstyle.org (or in your downloaded distribution).
+
+    To completely disable a check, just comment it out or delete it from the file.
+    To suppress certain violations please review suppression filters.
+
+    Authors: Max Vetrenko, Ruslan Diachenko, Roman Ivanov.
+ -->
```

Comment:
I am a bit conflicted about this comment. I understand we have picked up the google style xml file but should we be adding their comment or should we add our own and mention that we have picked it up from this particular xml file and reference it? @seanlip can you comment on this - this is more of a open source question?

Reply:
Could you or @anandwana001 please point me to the repo where this code snippet was taken from, and link to the original file from which these lines were taken? I think I need that info in order to weigh in.

Reply:
@seanlip From here - https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml

Reply:
OK, I did some reading and I think it would be better not to include an LGPL file as part of an Apache repo. Can you write a custom XML checkstyle for Oppia Android rather than using the default?

Also, in general, please never submit any code that's copied verbatim from other places to this repo. All code submitted in PRs must be original (including XML files).

Reply:
@seanlip  This checkstyle file (Google Java Style Guide) is something we need to follow as per Ben's comment - https://github.com/oppia/oppia-android/issues/1735#issuecomment-683076568 

If I create a custom, all I could do is remove comments, else all properties will remain the same as we want the same style to follow.

Reply:
Discussed offline. @anandwana001 will be creating a checkstyle file suitable for Oppia Android's purposes from scratch.

Reply:
Adding @BenHenning for reference, 

I was planning to use the google style config directly from the command which is one of the recommended as per the documentation, but the default is only WARN where it will not fail/error on checks. So, I had created this config file which is same as the google java style, only changed the property called severity to error so that checks fail. 

[discussed with the repo owner](https://github.com/checkstyle/checkstyle/issues/4452#issuecomment-727278218) if you could change param in the command, but it's not possible so I will be writing this file from fresh.

Reply:
It's probably preferable to have our own file, anyway, since it lets us ensure the enforce style exactly matches our style guide.

Reply:
Given that there's not really a reasonable way to derive our own version of the file without it more or less being a copy of the original, the easiest solution available to us seems to be to keep the default severity as warning rather than error. Given also that we plan to only keep Java around temporarily, I think we can live with this.

However one main thing we need to check: when Checkstyle fails with a warning, is the application exit code 0 or 1? If the latter, then CI will properly fail even if it's outputting warnings.

Reply:
With severity warning the CI will not fail.

Reply:
Does Checkstyle have any output if it succeeds? If not, could we do something like:

```bash
#!/bin/bash
lint_results=$(java -jar checkstyle-8.37-all.jar -c=checkstyle.xml app/src/ data/src/ domain/src/ utility/src/ testing/src/ 2>&1)
lint_command_result=$?
echo $lint_results
if [ $lint_command_result -ne 0] || [ -z "$lint_results" ]; then
  # Assume any lint output or non-zero exit code is a failure.
  exit 1
else
  exit 0
fi
```

This assumes any output is a fail. If the linter has regular output, then we might need to specifically check for 'warning' in the message. It does require that we put the linter command in a script, but we should be doing that, anyway, to avoid duplication with the configuration file.

Reply:
Done

Reply:
Just to check, have you confirmed that this correctly fails in CI with missing Java style issues checked in?

Reply:
@anandwana001 can you confirm this?

Reply:
yes, it fails correctly.

Reply:
ah, corrected, I was using sh earlier. fixed it now.

Reply:
Thanks for checking!